shared/asm-utils.fs

```diff
@@ -139,14 +158,13 @@
 \   - Quotient is put into BC
 \   - Remainder into HL
 : HL/DE,
-  D 7 # bit, #nz if,
+  D 7 # bit, AF push, #nz if,
     negate-DE,
-    HL/DE+,    \ BC = quotient, HL = remainder
+  then,
+  HL/DE+,    \ BC = quotient, HL = remainder
+  AF pop, #nz if,
     negate-BC,
-  else,
-    HL/DE+,
```

Comment:
It might be better to make `HL/DE+` a function to be called with `call`, but I'm not sure how to go about doing that. Can asm words call Forth words?

Reply:
If you just mean extracting it into an ASM subroutine, you can use `label foo` and `foo call,`. Calling a colon word from ASM is possible, but we currently don't support getting the address lazily from within a code definition.